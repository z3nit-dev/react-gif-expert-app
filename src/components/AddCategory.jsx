import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory })=>{
    const [ inputValue, setInputValue ] = useState('')
    
    const onInputChange = (event)=>{
        setInputValue(event.target.value)
    }

    const onSubmit = (event)=>{
        event.preventDefault()
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue)
        setInputValue('')
    }

    return (
        <form aria-label="form" onSubmit={onSubmit}>
            <input type="text" placeholder="Buscar categorias" value={inputValue} onChange={onInputChange}/>
        </form>
        
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}