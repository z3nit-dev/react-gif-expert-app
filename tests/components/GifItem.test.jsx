import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Prueba GifItem module', () => { 
    const url = 'https://google.es/'
    const title = 'Google'
    const moduleGifItem = <GifItem url={url} title={title}/>
    
    test('should Snapshot', () => { 
        const { container } = render(moduleGifItem)
        expect(container).toMatchSnapshot()
    })
    
    test('should mostrar la imagen con la URL y ALT indicado', () => { 
        render(moduleGifItem)
        
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('should mostrar el titulo en el parrafo', () => { 
        render(moduleGifItem)
        expect(screen.getByText(title)).toBeTruthy()
     })
 })