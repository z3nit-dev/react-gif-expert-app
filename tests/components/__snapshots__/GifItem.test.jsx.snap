// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba GifItem module should Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Google"
      src="https://google.es/"
    />
    <p>
      Google
    </p>
  </div>
</div>
`;
